src/bundle/Controller/User/ChangePasswordController.php

```diff
@@ -0,0 +1,112 @@
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Exception;
+
+class ChangePasswordController extends Controller
+{
+    /** @var NotificationHandlerInterface */
+    private $notificationHandler;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    /** @var LanguageService */
+    private $userService;
+
+    /** @var FormFactory */
+    private $formFactory;
+
+    /** @var TokenStorageInterface */
+    private $tokenStorage;
+
+    /**
+     * @param NotificationHandlerInterface $notificationHandler
+     * @param TranslatorInterface $translator
+     * @param UserService $userService
+     * @param FormFactory $formFactory
+     * @param TokenStorageInterface $tokenStorage
+     */
+    public function __construct(
+        NotificationHandlerInterface $notificationHandler,
+        TranslatorInterface $translator,
+        UserService $userService,
+        FormFactory $formFactory,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->notificationHandler = $notificationHandler;
+        $this->translator = $translator;
+        $this->userService = $userService;
+        $this->formFactory = $formFactory;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws \Symfony\Component\Translation\Exception\InvalidArgumentException
+     * @throws ContentValidationException
+     * @throws ContentFieldValidationException
+     * @throws InvalidOptionsException
+     */
+    public function changePasswordAction(Request $request): Response
+    {
+        $form = $this->formFactory->changePassword(
+            new ChangePasswordData()
+        );
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+
+            try {
+                $newPassword = $data->getNewPassword();
+                $userUpdateStruct = $this->userService->newUserUpdateStruct();
+                $userUpdateStruct->password = $newPassword;
+                $user = $this->tokenStorage->getToken()->getUser()->getAPIUser();
+
+                $this->userService->updateUser($user, $userUpdateStruct);
+
+                $this->notificationHandler->success(
+                    $this->translator->trans(
+                        /** @Desc("Your password has been successfully updated.") */
+                        'change_password.success',
+                        [],
+                        'change_password'
+                    )
+                );
+
+                return new RedirectResponse($this->generateUrl('ezplatform.dashboard'));
+            } catch (Exception $e) {
+                $this->notificationHandler->error($e->getMessage());
+            }
+        }
+
+        return $this->render('@EzPlatformAdminUi/user/change_password.html.twig', [
+            'form_change_password' => $form->createView(),
```

Comment:
With controller name change I proposed above I'd also introduce changed name in all variables, method names etc.